### ISSUE TITLE
How to use Transformer checkpoints for tensorflow graph execution

### ISSUE THREAD

**Issue by NehaPathapati** [tensorflow/tensor2tensor#882]
### Description
In the example show in Google Colab, tensorflow eager execution is used. However, can anyone provide an example to get tensorflow graph execution running with these checkpoints?
...

### Environment information

```
OS: <your answer here>

$ pip freeze | grep tensor
# your output here

$ python -V
# your output here
```

### For bugs: reproduction and error logs

```
# Steps to reproduce:
...
```

```
# Error logs:
...
```

**Comment by martinpopel** [comment]
Yes, the checkpoints downloaded from Google Colab can be used in T2T without eager mode. Just point `--output_dir` to the directory with the checkpoints when using `t2t-trainer` or `t2t-decoder` (and don't forget to download also the correct vocab file).

**Comment by NehaPathapati** [comment]
Thanks for the reply.

But, I would want to know how you would load checkpoints (English to German translation checkpoints already available) to a Tensorflow session and make inference from it?

**Comment by martinpopel** [comment]
I don't understand your question. There is [t2t-decoder](https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/bin/t2t_decoder.py) for this purpose. You can use it also as a Python module, just provide the appropriate FLAGS. Alternatively, you can use `decoding.decode_from_file` as done in `t2t_decoder.py`.

**Comment by NehaPathapati** [comment]
latest_checkpoint_path = tf.train.latest_checkpoint(os.path.join(PATH_TO_CHECKPOINTS_DIRECTORY))

 with tf.Session() as sess:
        tf_saver = tf.train.Saver()
        model = tf_saver.restore(sess, latest_checkpoint_path)

So, PATH_TO_CHECKPOINTS_DIRECTORY is the path to the downloaded checkpoints for English to German translation. 
I have then restored it using tf.saver.restore.

I would like to know how I can use this for inference/translation.

**Comment by RayXu14** [comment]
I have same question. I want to load checkpoint and use tensor2tensor to make inference, without enabling eager execution. So can tensor2tensor works without enabling eager execution?

**Comment by RayXu14** [comment]
no, I already give up tensor2tensor

**Comment by ajallooeian** [comment]
I am having the same issue. I have trained a model using t2t cli, but now I want to use it for "non eager" inference in a -->python<-- code. Did someone find the solution in the end?

**Comment by Santosh-Gupta** [comment]
I have a similar issue. I want to continue training from a checkpoint. It seems that I just have to put it in my `--output_dir` and it'll pick up on it?

What if I want to start training from a checkpoint, but save to a different checkpoint?

**Comment by martinpopel** [comment]
Yes, by default training resumes from the last checkpoint stored in `--output_dir` (more precisely, there is file `checkpoint` which specifies the "last" checkpoint).
T2T adds the number of steps to the checkpoint name, so it will never overwrite an existing checkpoint. So saving to a different checkpoint is the default.
